Add Main UI docs for integration into sidebar help

florian-h05 opened the pull request:
Required for https://github.com/openhab/openhab-webui/pull/2253.

/cc @stefan-hoehn @Confectrician 

Please have a look at the CONTRIBUTING.md inside the mainui folder.

florian-h05 commented:
I have now migrated all content that was hard-coded into the UI to the docs repo and created the structure for the new Main UI docs. You can run my linked UI PR locally to see how the UI integrates the docs.

Feel free to extend the documentation, migrate docs from elsewhere in this repo and put some content into the empty files.

stefan-hoehn commented on mainui/settings/scenes.md:
```diff
@@ -0,0 +1,26 @@
+  Use the <!--F7 pause_circle --> pause buttons on the individual scene pages.
+- **Manually run Scenes**<br>
+  Use the <!--F7:blue play --> play button on the individual scene pages to have the scene run immediately.
+<!-- END MAINUI SIDEBAR DOC - DO NOT REMOVE -->
```
```
<!-- END MAINUI SIDEBAR DOC - DO NOT REMOVE -->

A comprehensive introduction to scenes can be found at the 
[scenes tutorial]({{base}}/tutorial/rules_scenes.html).
```

stefan-hoehn commented on mainui/settings/schedule.md:
```diff
@@ -0,0 +1,39 @@
+
+![timer-trigger](../images/timer-trigger.png)
+
+this rule will be shown in the calendar view on this page.
```
then this

stefan-hoehn commented on mainui/settings/schedule.md:
```diff
@@ -0,0 +1,39 @@
+
+![saturday-morning-rule](../images/saturday-rule-schedule.png)
+
+A rule that is schedule repeatedly like everyday at 8:00 in the morning will therefore be shown on every day of the calendar view.
```
A rule that is scheduled repeatedly

stefan-hoehn commented on mainui/settings/scripts.md:
```diff
@@ -0,0 +1,25 @@
+<!-- START MAINUI SIDEBAR DOC - DO NOT REMOVE -->
+**Rules** are the heart of home automation.
+**Scripts** are special types of rules with only one action and no triggers, so they have to be called from "real" rules or run manually.
+They can be used to reuse code and logic across multiple other rules, one off experimentation to figure something out, create a catalog of examples, or to drive tests for other rules and behaviors.
```
for experimentation to figure

stefan-hoehn commented on mainui/settings/things.md:
```diff
@@ -0,0 +1,40 @@
+- **Add new Things**<br>
+  Use the [Thing Inbox](#inbox) to scan for available things or manually add ones that cannot be found automatically using the <!--F7:blue plus_circle_fill --> plus button.
+- **Configure Things**<br>
+  Click on any Thing in the list to see its configuration page and it's channels.
```
its channels.

stefan-hoehn commented on mainui/settings/things.md:
```diff
@@ -0,0 +1,40 @@
+It therefore makes a lot of sense to use these features for a smart home solution.
+
+openHAB bindings can therefore implement Discovery Services for Things, which provide Discovery Results.
+All Discovery Results are put into the Thing inbox, from where they can be easily added to openHAB with just a few clicks.
```
Thing inbox from where they can be

stefan-hoehn commented on mainui/settings/transformations.md:
```diff
@@ -0,0 +1,22 @@
+# Settings - Transformations
+
+<!-- START MAINUI SIDEBAR DOC - DO NOT REMOVE -->
+**Transformations** convert a value or Item state into different value or state.
```
into a different value

stefan-hoehn reviewed (changes requested):
@florian-h05 This is an amazing contribution to the docs and you earn the docs gold medal for the month 🥇 🤩

Given the amount of text you contributed I only found only a few minor issues.

florian-h05 commented on mainui/CONTRIBUTING.md:
```diff
@@ -0,0 +1,50 @@
+Inside these comments, the UI allows the following features:
+
+- Embed [Framework7 icons](https://framework7.io/icons/) into the UI-rendered docs using these comments:
+  - `<!--F7 $iconame -->`, where `$iconname` is a Framework7 icon's name
```
It is one n missing in `$iconame` ;-)

florian-h05 commented on mainui/settings/persistence.md:
```diff
@@ -0,0 +1,25 @@
+openHAB by default ships with the [RRD4J persistence service](/addons/persistence/rrd4j/) installed by default, which persists every Item on every state change and at least once a minute.
+Additionally, it restores the last stored value at system startup.
+
+To change this behaviour, you already found the correct page!
```
![image](https://github.com/openhab/openhab-docs/assets/73423173/c2bc26fe-1238-47e6-a84c-afbf033df2d8)

florian-h05 commented on mainui/settings/rules.md:
```diff
@@ -0,0 +1,55 @@
+  Use the <!--F7:blue play --> play button on the individual rule pages to have the rule run all actions immediately.
+<!-- END MAINUI SIDEBAR DOC - DO NOT REMOVE -->
+
+The page has a searchable list of rules where the following attributes
```
Great that you review your own contribution ;-)
I have only moved that from the old location to the new location, I haven't written this.

florian-h05 commented:
@stefan-hoehn Thank you for your fast review of this big PR!! I have addressed your review.

> This is an amazing contribution to the docs and you earn the docs gold medal for the month 🥇 🤩

Thank you very much, I guess I will share it with you since you wrote large parts of the docs that I moved around, and with @JustinGeorgi, who originally wrote what has been added to the UI and what I migrated to the docs repo.

stefan-hoehn reviewed (approved):
Thanks, looks good now.

florian-h05 commented:
@ghys @Confectrician Why can’t I find that documentation changes on https://next.openhab.org/docs?

ghys commented:
The website fails to build on Netlify since Jan 26th:
https://app.netlify.com/sites/openhab-next/deploys/65c1516267c7c40008971007
with this error:
```
10:22:45 PM: /opt/buildhome/.rvm/rubies/ruby-2.4.3/lib/ruby/2.4.0/fileutils.rb:1225:in `stat": No such file or directory @ rb_file_s_stat - .vuepress/openhab-docs/settings/images (Errno::ENOENT)
	from /opt/buildhome/.rvm/rubies/ruby-2.4.3/lib/ruby/2.4.0/fileutils.rb:1225:in `lstat"
	from /opt/buildhome/.rvm/rubies/ruby-2.4.3/lib/ruby/2.4.0/fileutils.rb:1254:in `copy"
	from /opt/buildhome/.rvm/rubies/ruby-2.4.3/lib/ruby/2.4.0/fileutils.rb:416:in `block in copy_entry"
	from /opt/buildhome/.rvm/rubies/ruby-2.4.3/lib/ruby/2.4.0/fileutils.rb:1388:in `wrap_traverse"
	from /opt/buildhome/.rvm/rubies/ruby-2.4.3/lib/ruby/2.4.0/fileutils.rb:413:in `copy_entry"
	from /opt/buildhome/.rvm/rubies/ruby-2.4.3/lib/ruby/2.4.0/fileutils.rb:391:in `block in cp_r"
	from /opt/buildhome/.rvm/rubies/ruby-2.4.3/lib/ruby/2.4.0/fileutils.rb:1461:in `block in fu_each_src_dest"
	from /opt/buildhome/.rvm/rubies/ruby-2.4.3/lib/ruby/2.4.0/fileutils.rb:1477:in `fu_each_src_dest0"
	from /opt/buildhome/.rvm/rubies/ruby-2.4.3/lib/ruby/2.4.0/fileutils.rb:1459:in `fu_each_src_dest"
	from /opt/buildhome/.rvm/rubies/ruby-2.4.3/lib/ruby/2.4.0/fileutils.rb:390:in `cp_r"
	from prepare-docs.rb:359:in `<main>"
```
corresponding to this line in prepare-docs.rb:
https://github.com/openhab/website/blob/31c746e62fa4c3d5a3fc33c070713acfd2dd308c/prepare-docs.rb#L359

florian-h05 commented:
https://github.com/openhab/website/pull/445 will fix the website build.
